src/auth/basic/core/qgsauthbasicmethod.cpp

```diff
@@ -139,6 +139,21 @@ bool QgsAuthBasicMethod::updateDataSourceUriItems( QStringList &connectionItems,
   // Branch for OGR
   if ( dataprovider == QLatin1String( "ogr" ) || dataprovider == QLatin1String( "gdal" ) )
   {
+    // If username or password contains comma or double quote we need to quote the string
```

Comment:
I'm afraid this logic won't work with all OGR drivers, and should probably be conditionalized to the MySQL one, given that code below for the PG driver uses single quotes

Reply:
Ok, but the question is:

Is it documented somewhere?

Reply:
> Ok, but the question is:
> 
> Is it documented somewhere?

for PG at least: https://gdal.org/drivers/vector/pg.html#connecting-to-a-database

Reply:
Ok done, I suppose we should implement a similar logic for the other drivers but as far as this issue is concerned it is fixed.